packages/headless/src/database/mongodb/order/index.ts

```diff
@@ -1,32 +1,78 @@
-import { 
-  ChangeStatusEntity, 
-  GetAllOrderQueryEntity, 
-  OrderEntity, 
-  OrderIncompleteStatEntity, 
-  OrderSortQuery, 
-  OrderStatEntity, 
-  OrderStatusEnum, 
-  ShippingStatusEnum, 
-  StatusTypeDto 
+import {
+  CartItem, 
+  ReOrderQuery,
+  ChangeStatusEntity,
+  GetAllOrderQueryEntity,
+  OrderEntity,
+  OrderIncompleteStatEntity,
+  OrderSortQuery,
+  OrderStatEntity,
+  OrderStatusEnum,
+  ProductOrder,
+  ShippingStatusEnum,
+  StatusTypeDto,
 } from 'src/entity/order';
 import { IOrderDatabase } from 'src/modules/order/repositories/order.db.interface';
 import { ProductModel } from '../product/product.model';
 import { OrderModel } from './order.model';
-import { CreateOrderRequest, CreateProductOrderDetails, IProductOrderData } from 'models';
+import {
+  CreateOrderRequest,
+  CreateProductOrderDetails
+} from 'models';
+import { CartModel } from '../cart/cart.model';
 
 export class OrderDatabase implements IOrderDatabase {
+  async addToCart(userId: string, items: CartItem[]) : Promise<any> {
+    let addItems;
+    try{
+      addItems = await Promise.all(items.map(async (item) => await CartModel.findOneAndUpdate( { userId },{ $push: { items: item } },{ new: true } )));
+    }catch(err){
+      console.log(err);
+      return null;
+    }
+    return addItems;
+  }
+
+  async getCart(userId: string): Promise<any> {
+    return await CartModel.findOne({userId }).select('-_id').lean();
+  }
   async createOrder(userId: string, body: CreateOrderRequest): Promise<OrderEntity> {
     return await OrderModel.create({ userId, ...body });
   }
 
-  async addPhotoDetails(products: CreateProductOrderDetails[]): Promise<IProductOrderData[]>{
+  async getAvailableProducts(products: ProductOrder[]): Promise<ProductOrder[]>{
+    let items = [];
+    await Promise.all(products.map( async (product) => {
+        const available = await ProductModel.findOne({id: product.productId}).lean();
+        if(!available){
+          items.push(product);
+        }
+        return product;
+    }));
```

Comment:
https://www.mongodb.com/docs/v4.2/tutorial/query-documents/
read the docs and update your query